AutomatedTesting/Gem/PythonTests/editor/EditorScripts/Sample_Editor_Tests.py

```diff
@@ -0,0 +1,143 @@
+                 False if entity id does not exist in the entity_list
+        """
+        entity_list = []
+        entity_search_filter = entity.SearchFilter()
+        entity_search_filter.names = entity_name
+        entity_list = entity.SearchBus(bus.Broadcast, 'SearchEntities', entity_search_filter)
+        if entity_list:
+            if entity_to_search in entity_list:
+                return True
+            return False
+        return False
+
+    # 01. create_level
+
+    test_level = 'Simple'
+    general.open_level_no_prompt(test_level)
+    Report.result(Tests.create_level, general.get_current_level_name() == test_level)
+
+    # 02. load existing level - skipping this since this could alter existing level that other test depends on
+
+    # 03. create_entity and set name
+    # Delete any exiting entity and Create a new Entity at the root level
+    search_filter = azlmbr.entity.SearchFilter()
+    all_entities = entity.SearchBus(azlmbr.bus.Broadcast, "SearchEntities", search_filter)
+    editor.ToolsApplicationRequestBus(bus.Broadcast, "DeleteEntities", all_entities)
+    parent_entity = editor.ToolsApplicationRequestBus(bus.Broadcast, "CreateNewEntity", entity.EntityId())
+    Report.result(Tests.create_entity, parent_entity.IsValid())
+
+    # Setting a new name
+    parent_entity_name = "Parent_1"
+    editor.EditorEntityAPIBus(bus.Event, 'SetName', parent_entity, parent_entity_name)
+    Report.result(Tests.set_entity_name,
+                  editor.EditorEntityInfoRequestBus(bus.Event, 'GetName', parent_entity) == parent_entity_name)
+
+    # 04. Creating child Entity and setting name to above created parent entity
+    child_1_entity = editor.ToolsApplicationRequestBus(bus.Broadcast, 'CreateNewEntity', parent_entity)
+    Report.result(Tests.create_child_entity, child_1_entity.IsValid())
+    child_entity_name = "Child_1"
+    editor.EditorEntityAPIBus(bus.Event, 'SetName', child_1_entity, child_entity_name)
+    Report.result(Tests.set_entity_name,
+                  editor.EditorEntityInfoRequestBus(bus.Event, 'GetName', child_1_entity) == child_entity_name)
+
+    # 05. delete_Child_entity
+    editor.ToolsApplicationRequestBus(bus.Broadcast, 'DeleteEntityById', child_1_entity)
+    Report.result(Tests.delete_entity, search_entity(child_1_entity, "Child_1") == False)
+
+    # 06. add mesh component to parent entity
+    type_id_list = editor.EditorComponentAPIBus(bus.Broadcast, 'FindComponentTypeIdsByEntityType', ["Mesh"],
+                                                entity.EntityType().Game)
+    if type_id_list is not None:
+        component_outcome = editor.EditorComponentAPIBus(bus.Broadcast, 'AddComponentsOfType', parent_entity,
+                                                         type_id_list)
+        Report.result(Tests.add_mesh_component, component_outcome.IsSuccess())
+    else:
+        Report.result(Tests.found_component_typeId, type_id_list is not None)
+
+    # 09. remove mesh component
```

Comment:
This is the second instance of only running some tests given failure conditions. We should utilize the pass/fail nature of the Tests class to report the status rather than only ever trigger the pass status of one test or failure status of another. Additionally, a lot of this code is already available in utilities, which can be utilized so that if something needs an update, it only needs to be updated in one place:

```
    import editor_python_test_tools.hydra_editor_utils as hydra

    # 09. Remove Mesh component
    hydra.remove_component('Mesh', parent_entity)
    removal_success = not hydra.has_components(parent_entity, ['Mesh'])
    Report.result(Tests.remove_mesh_component, removal_success)

Reply:
I'd point to .\o3de\AutomatedTesting\Gem\PythonTests\EditorPythonTestTools\editor_python_test_tools\editor_entity_utils.py but it doesn't have a remove_component at this point...
second the idea that you should lean on central utils for as much as you can. if they are missing reusable functions like remove_component, you can look to add then. only rarely should test code actually have raw bus calls in it, specifically when its a one off bus call that is unlikely to ever be used again.

Reply:
i choose not to use utils to so that I can develop an understanding of how the core hyda functions work. Is it a must to use utils?

Reply:
fixed in latest commit

Reply:
It is not a must, but it is extremely useful. If something changes in any of these bus calls, there would be maintenance required on both the utility that already does this AND your script. Utilizing the utils ensures we only need to maintain one area. It's definitely good to understand what's going on behind the scenes with practice, but what gets checked in should be as maintainable as possible.